store/db.go

```diff
@@ -44,39 +48,49 @@ func NewDB(dbdir string) (*DB, error) {
 
 func (db *DB) Open() error {
 	// take a lock on db dir
-	if db.lock != nil {
-		panic("cas db lock already gained")
-	}
 	dl, err := lock.ExclusiveLock(db.dbdir, lock.Dir)
 	if err != nil {
 		return err
 	}
-	db.lock = dl
 
 	sqldb, err := sql.Open("ql", filepath.Join(db.dbdir, DbFilename))
 	if err != nil {
 		dl.Close()
 		return err
 	}
+
+	// (XXX): flock is already granted, the mutex
```

Comment:
This doesn't sit right. The flock and mutex have different purposes: the former to prevent _process_ races, the latter for _thread_ (goroutine) races.